### PR TITLE
Issue #596 Adjusting headers of generated adoc files and updates to site

### PR DESCRIPTION
**Pull request by hferentschik**
This addresses #596

### CODE REVIEW

**Comment by hferentschik** [comment]
@budhrg can you try this and verify the tarball with the doc.openshift.org build? It is now also documented in this pull request how to do this. If all works we should merge this asap.

Once merged, we should also upload the tarball generated by this to the artifacts server. Either manually or via the Jenkins job.

@LalatenduMohanty this are the changes required to make the docs properly build in docs.openshift.org. Please try as well. 

@thatdocslady This should make things works from a an integration point of view. Still some more docs changes needed.

File: docs/source/developing/writing-docs.adoc [hunk 1]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
 
+[[section-deploying-minishift-docs]]
 == Deploying Minishift Documentation
 
-TBD
+The Minishift documentation is deployed on _docs.openshift.org_ under
+link:https://docs.openshift.org/minishift[https://docs.openshift.org/minishift].
+
+To integrate with the _docs.openshift.org_ we are delivering a tarball containing
+the Minishift asciidoc files as well as some link:http://www.asciibinder.org/[AsciiBinder].
```
Comment (budhrg) [inline docs/source/developing/writing-docs.adoc]: extra full stop at the end of line 58

File: docs/source/developing/writing-docs.adoc [hunk 2]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
+<<section-building-openshift-docs>>.
+
+[[section-building-openshift-docs]]
+=== Building OpenShift documentation
+
+To start with you need a checkout of the link:https://github.com/openshift/openshift-docs.git[openshift-docs]
+GitHub repository. You need all the tooling to build openshift-docs. Refer to
+link:https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/tools_and_setup.adoc[Install and set up the tools and software] which is part of the openshift-docs repository.
+
+Once you can successfully run `rake build` in your openshift-docs checkout, you can integrate the
```
Comment (budhrg) [inline docs/source/developing/writing-docs.adoc]: Once you can successfully run `rake build` in your openshift-docs checkout, you can able to integrate the Minishift documentation.

Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: " you can able to" ?

File: docs/source/developing/writing-docs.adoc [hunk 3]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
+$ mkdir minishift
+$ cd minishift
+$ cp <path to tarball> .
+$ tar -xvf minishift-adoc.tar --strip 1
+$ cat _topic_map.yml >> ../_topic_map.yml
+$ cd ..
+$ rake build
+----
+
+If the site build completes successfully, the site is available under _preview/openshift-origin/latest/welcome/index.html_.
```
Comment (budhrg) [inline docs/source/developing/writing-docs.adoc]: If the build completes successfully, the site is available under ....

File: docs/source/developing/writing-docs.adoc (hunk 1 above)
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: fixed, thanks

File: docs/source/developing/writing-docs.adoc (hunk 3 above)
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: ok

File: docs/source/developing/developing.adoc [hunk 4]
```diff
@@ -20,7 +20,7 @@ Mac OS X or Windows. The Windows operating system might require additional steps
 have some limitations.
 
 [[set-up-dev-env]]
-== Setting up the development environment
+== Workspace creation
```
Comment (LalatenduMohanty) [inline docs/source/developing/developing.adoc]: Actually "Setting up the development environment" sounds better to me i.e. I mean may be in India we use it more. @Preeticp comments?

Comment (hferentschik) [inline docs/source/developing/developing.adoc]: fair enough

**Comment by budhrg** [comment]
Hi @hferentschik , awesome job!

All the content, layout seems to working as expected except a bit of gap is need in few places. See below
#### Existing and expected Gap
expected_gap
 
#### Actual Gap
actual_gap

**Comment by hferentschik** [comment]
> All the content, layout seems to working as expected except a bit of gap is need in few places. See below

I don't think we have to worry about this, right now. :-) Thanks for checking and the review.

File: docs/source/developing/writing-docs.adoc [hunk 5]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
+
+[source,sh]
+----
+$ make gen_adoc_tar
+----
+
+Once the build completes there will be a _minishift-adoc.tar_ under _docs/build_.
+
+How this tarball is integrated into the docs.openshift.org site build is described in
+<<section-building-openshift-docs>>.
```
Comment (Preeticp) [inline docs/source/developing/writing-docs.adoc]: The integration of this tarball is described in the following section. 
We probably do not need a link here, it is the immediate next section.

File: docs/source/developing/writing-docs.adoc [hunk 6]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
+
+[[section-building-openshift-docs]]
+=== Building OpenShift documentation
+
+To start with you need a checkout of the link:https://github.com/openshift/openshift-docs.git[openshift-docs]
+GitHub repository. You need all the tooling to build openshift-docs. Refer to
```
Comment (Preeticp) [inline docs/source/developing/writing-docs.adoc]: Maybe we could break this out into steps:
1. Checkout the link:https://github.com/openshift/openshift-docs.git[openshift-docs]
 +GitHub repository.
2. Install all the tooling required to build the OpenShift-docs. For further information refer to +link:https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/tools_and_setup.adoc[Install and set up the tools and software] section of the openshift-docs repository.
3. You can integrate the Minishift documentation once you are able to successfully run the `rake build` in your openshift-docs checkout.

File: docs/source/developing/writing-docs.adoc [hunk 7]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
 port 4567. You can access the rendered documentation by browsing to
 http://<IP of Docker daemon>:4567.
 
+[[section-deploying-minishift-docs]]
 == Deploying Minishift Documentation
```
Comment (Preeticp) [inline docs/source/developing/writing-docs.adoc]: The second level titles seem to be using Title case, whereas the third level ones use Sentence case, not sure if this is the standard we follow from the OpenShift docs.

File: docs/source/developing/ci.adoc [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+toc::[]
+
+The following sections describes the various CI setups Minishift utilizes.
+
+[[ci-setup]]
+== CentOS CI
+
+Minishift uses several CI build servers, including https://ci.centos.org/[CentOS CI]. The server
+builds incoming pull requests and pushes to master. Is also archives build artifacts of successful
```
Comment (hferentschik) [inline docs/source/developing/ci.adoc]: +1

File: docs/source/developing/writing-docs.adoc [hunk 9]
```diff
@@ -25,23 +20,26 @@ maintained https://en.wikipedia.org/wiki/AsciiDoc[Asciidoc] files.
 Per default the documentation is build in a Docker container. This way
 you avoid having to install all required dependencies on your
```
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: ok

File: docs/source/developing/writing-docs.adoc [hunk 10]
```diff
@@ -25,23 +20,26 @@ maintained https://en.wikipedia.org/wiki/AsciiDoc[Asciidoc] files.
...
 development machine. All you need is a running Docker daemon. In case
-you don't have one, use Minishift itself. See (TODO reference to the
-Minishift as Docker daemon from README. Needs to be extracted).
+you don't have one, use Minishift itself. See also
+link:../command-ref/minishift_docker-env{outfilesuffix}[`minishift docker-env`].
```
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: Ok for now. Once the other outstanding pull request is merged, I'll repoint the link anyways.

File: docs/source/developing/writing-docs.adoc (hunk 7 above)
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: I'll aligned it

File: docs/source/developing/writing-docs.adoc [hunk 11]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
 http://<IP of Docker daemon>:4567.
 
+[[section-deploying-minishift-docs]]
 == Deploying Minishift Documentation
 
-TBD
+The Minishift documentation is deployed on _docs.openshift.org_ under
+link:https://docs.openshift.org/minishift[https://docs.openshift.org/minishift].
+
+To integrate with the _docs.openshift.org_ we are delivering a tarball containing
```
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: +1

File: docs/source/developing/writing-docs.adoc [hunk 12]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
+the Minishift asciidoc files as well as some link:http://www.asciibinder.org/[AsciiBinder]
+meta data.
+
+This tarball can be build via:
+
```
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: ok

File: docs/source/developing/writing-docs.adoc [hunk 13]
```diff
@@ -50,6 +48,51 @@ The latter will start the https://middlemanapp.com[Middleman] server on
...
+[source,sh]
+----
+$ make gen_adoc_tar
+----
+
+Once the build completes there will be a _minishift-adoc.tar_ under _docs/build_.
+
```
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: I changed this a bit

File: docs/source/developing/writing-docs.adoc (hunk 6 above)
Comment (hferentschik) [inline docs/source/developing/writing-docs.adoc]: I don't think this is required.

**Comment by budhrg** [comment]
Rest looks good to me 👍  and agree we can make space adjustments later on too

File: docs/source/developing/writing-docs.adoc (hunk 6 above)
Comment (LalatenduMohanty) [inline docs/source/developing/writing-docs.adoc]: +1 for numbered list as the steps are sequential.

**Comment by hferentschik** [comment]
@Preeticp thanks for your review. Applied most of your suggestions.

**Comment by Preeticp** [comment]
Welcome @hferentschik

**Comment by thatdocslady** [comment]
FYI the spacing issue is probably because there is no H2 title after the TOC (you can see "Overview" in the OpenShift topic). We can open a follow-up issue for that and add these titles if the spacing really annoys us :)

**Pull request merged**